src/txn/txn_rollback_to_stable.c

```diff
@@ -784,12 +842,14 @@ __rollback_page_needs_abort(
         __wt_cell_unpack_addr(session, ref->home->dsk, (WT_CELL *)addr, &vpack);
         durable_ts = __rollback_get_ref_max_durable_timestamp(session, &vpack.ta);
         prepared = vpack.ta.prepare;
-        result = (durable_ts > rollback_timestamp) || prepared;
+        result = (durable_ts > rollback_timestamp) || prepared ||
+          WT_ROLLBACK_CHECK_RECOVERYFLAG_TXNID(session, vpack.ta.newest_txn);
     } else if (addr != NULL) {
         tag = "address";
         durable_ts = __rollback_get_ref_max_durable_timestamp(session, &addr->ta);
         prepared = addr->ta.prepare;
-        result = (durable_ts > rollback_timestamp) || prepared;
+        result = (durable_ts > rollback_timestamp) || prepared ||
```

Comment:
This macro can be used `WT_ROLLBACK_CHECK_RECOVERYFLAG_TXNID` instead of repeating the same code.

Reply:
Updated to use WT_ROLLBACK_CHECK_RECOVERYFLAG_TXNID flag and fixed failing tests.